lib/clients/GcpClient.py

```diff
@@ -15,7 +15,12 @@ def __init__(self, operation_name, configuration, directory_persistent, director
         super(GcpClient, self).__init__(operation_name, configuration, directory_persistent, directory_work_list,
                                         poll_delay_time, poll_maximum_time)
 
-        self.__gcpCredentials = json.loads(configuration['credentials'])
+        if not 'credhub_url' in configuration:
+            self.__gcpCredentials = json.loads(configuration['credentials'])
+        else:
+            self.__gcpCredentials = self._get_credentials_from_credhub(
```

Comment:
Please check that this returns credentials in required format as for GCP it is in json while other IaaS are simple key-value pairs.

Reply:
Also, same as above. What happens if there is an error in getting these credentials? Do we need to raise exception explicitly?